### ISSUE TITLE
Any incoming message while waiting for sync subprocess treated as error

### ISSUE THREAD

**Issue by nwalker** [saleyn/erlexec#54]
```
(kerl:r17) ~/p/erlexec (master|✔)                                                                                                                                       
> erl -pa ebin
Erlang/OTP 17 [erts-6.4] [source] [64-bit] [smp:8:8] [async-threads:10] [kernel-poll:false]

Eshell V6.4  (abort with ^G)
1> application:start(exec).
ok
2> application:which_applications().
[{exec,"OS Process Manager","v1.0-156-g05edfd2"},
 {stdlib,"ERTS  CXC 138 10","2.4"},
 {kernel,"ERTS  CXC 138 10","3.2"}]
3> spawn(fun () -> io:format("~p~n", [exec:run("sleep 3 && echo done", [sync, stdout])]) end) ! test.
test
{error,[{reason,test}]}
```
As I can see, there is a [clause in exec:wait_for_ospid_exit](https://github.com/saleyn/erlexec/blob/master/src/exec.erl#L724-L725) which treats any incoming message as an error message. Are you sure this behaviour is correct?

**Comment by saleyn** [comment]
This function was added by one of the contributors, and all side-effects may not have been fully tested.

The function is reporting that it doesn't recognize the `'test'` message, which is correct.  Though, it's questionable if in that case the waiting process needs to be crashed or handle it the way it does.

**Comment by 0xNaN** [comment]
Hi, I'm dealing with what I think could be a strange behaviour (but I'm not really sure though!) and I think could be linked to this issue.
When running a process without linking to it and then killing it in a sync way, works without problems:
```
(3334@workstation)1> exec:run("sleep 100", []).
{ok,<0.105.0>,19393}
(3334@workstation)2> exec:run("kill 19393", [sync]).
{ok,[]}
```
But if the shell `trap_exit` and the first process is linked to it:

```
(3334@workstation)3> process_flag(trap_exit, true). 
false
(3334@workstation)4> exec:run_link("sleep 100", []).
{ok,<0.110.0>,19686}
(3334@workstation)5> exec:run("kill 19686", [sync]).
{error,[{reason,{'EXIT',<0.110.0>,{exit_status,15}}}]}
(3334@workstation)6> flush().
Shell got {'DOWN',#Ref<0.0.2.167>,process,<0.112.0>,normal}
ok
(3334@workstation)7> 
```
the `kill` process on `5>` dies with an error that is the `EXIT` of the `sleep` process (that I expected to trap) while the shell traps a `DOWN` (which I think became from the iternal of  `5>`);

What I'm expecting is that `5>` evaluates simply to `{ok,[]}` ( since `kill` exits with `0`) and the shell traps the `EXIT` message.

I'm missing something?

**Comment by saleyn** [comment]
Is it possible you are not running the latest version of erlexec?

Here's what I get when I execute your example on the master branch of the project:
```
(mta@nova)18> process_flag(trap_exit, true).                    
true
(mta@nova)19> f(P), {ok, _, P} = exec:run_link("sleep 100", []).
{ok,<0.6702.0>,27603}
(mta@nova)20> exec:run("kill " ++ integer_to_list(P), [sync]).  
{ok,[]}
(mta@nova)21> flush().
Shell got {'EXIT',<0.6702.0>,{exit_status,15}}
ok
```

**Comment by nwalker** [comment]
Issue already fixed in master by 95a9f21 / #58

**Comment by 0xNaN** [comment]
yes I was using an old version I'm really sorry!
I was using erlexec through rebar3 simply using `{deps, [erlexec]}` (therefore using the latest version available on `hex.pm` I think).
Using this repo solves my issue! anyway thank you so much :smile:
